data-index/data-index-storage/data-index-storage-jpa-common/src/main/java/org/kie/kogito/index/jpa/storage/ProcessInstanceEntityStorage.java

```diff
@@ -174,13 +194,11 @@ private NodeInstanceEntity updateNode(NodeInstanceEntity nodeInstance, ProcessIn
         return nodeInstance;
     }
 
-    private void indexSLA(ProcessInstanceSLAEventBody data) {
-        findOrInit(data.getProcessId(), data.getProcessInstanceId(), data.getEventDate());
-        repository.flush();
+    private void indexState(ProcessInstanceEntity pi, ProcessInstanceStateDataEvent event) {
+        indexState(pi, event.getData(), event.getKogitoAddons() == null ? Set.of() : Set.of(event.getKogitoAddons().split(",")), event.getSource() == null ? null : event.getSource().toString());
```

Comment:
You could also consider that `event.getKogitoAddons().isEmpty())` is empty for the ternary operator.
```suggestion
        indexState(pi, event.getData(), (event.getKogitoAddons() == null || event.getKogitoAddons().isEmpty()) ? Set.of() : Set.of(event.getKogitoAddons().split(",")), event.getSource() == null ? null : event.getSource().toString());
```